extensions/data-plane/data-plane-api/src/main/java/org/eclipse/dataspaceconnector/dataplane/api/transfer/DataPlanePublicApiRequestFilter.java

```diff
@@ -90,36 +86,46 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
             return;
         }
 
-        // create request and perform validation
-        var dataFlowRequest = createDataFlowRequest(tokenValidationResult.getContent(), requestContext);
-        var validationResult = dataPlaneManager.validate(dataFlowRequest);
-        if (validationResult.failed()) {
-            requestContext.abortWith(validationResult.getFailureMessages().isEmpty() ?
-                    validationError(String.format("Failed to validate request with id: %s", dataFlowRequest.getId())) :
-                    validationErrors(validationResult.getFailureMessages()));
-            return;
-        }
+        // create the data flow request and map it into the request body
+        var body = new BufferedReader(new InputStreamReader(requestContext.getEntityStream()))
+                .lines().collect(Collectors.joining("\n"));
+        var properties = createRequestProperties(requestContext.getRequest(), requestContext.getUriInfo(), requestContext.getMediaType(), body);
+        var dataFlowRequest = createDataFlowRequest(tokenValidationResult.getContent(), properties);
+        var serialized = typeManager.writeValueAsString(dataFlowRequest);
+        requestContext.setEntityStream(new ByteArrayInputStream(serialized.getBytes()));
+
+        // redirect to endpoints exposed by the controller
+        requestContext.setRequestUri(URI.create(requestContext.getUriInfo().getBaseUri() + publicApiPath));
+        requestContext.getHeaders().put(CONTENT_TYPE, List.of(MediaType.APPLICATION_JSON));
+    }
 
-        // perform the data transfer
-        var stream = new ByteArrayOutputStream();
-        var sink = new OutputStreamDataSink(stream, executorService, monitor);
-        var transferResult = dataPlaneManager.transfer(sink, dataFlowRequest)
-                .exceptionally(throwable -> TransferResult.failure(ResponseStatus.FATAL_ERROR, "Unhandled exception: " + throwable.getMessage()))
-                .join();
-        if (transferResult.failed()) {
-            requestContext.abortWith(internalErrors(transferResult.getFailureMessages()));
-            return;
+    /**
+     * Map relevant information from the container request context to the properties of the {@link DataFlowRequest}.
+     */
+    private Map<String, String> createRequestProperties(Request request, UriInfo uriInfo, @Nullable MediaType mediaType, String body) {
+        var props = new HashMap<String, String>();
+        props.put(METHOD, request.getMethod());
+        props.put(QUERY_PARAMS, convertQueryParamsToString(uriInfo));
+
+        if (!uriInfo.getPathSegments().isEmpty()) {
+            props.put(PATH, uriInfo.getPathSegments().stream()
```

Comment:
very neat. However, the publicApiPath could me multi-segmented too? (stuff/moreStuff) So it could not be compared against a single segment of the request uri, doesn't it? Maybe just hypothetical but this should be documented somewhere.

Reply:
Very good remark. 
I updated to something more resilient and add a unit-test for this scenario.